mllib/src/main/scala/org/apache/spark/mllib/feature/IDF.scala

```diff
@@ -218,7 +218,7 @@ private object IDFModel {
           newValues(k) = values(k) * idf(indices(k))
           k += 1
         }
-        Vectors.sparse(n, indices, newValues)
+        Vectors.sparse(n, indices, newValues).toSparse
```

Comment:
That version of the factory method would also work here; it would just require an `unzip`. Curious: why is one preferred over the other?